src/test/java/org/jenkinsci/plugins/workflow/support/concurrent/TimeoutTest.java

```diff
@@ -54,7 +53,14 @@
         } catch (InterruptedException x) {
             // good
         }
-        Thread.sleep(1_000);
+    }
+
+    @Test(expected = InterruptedException.class)
+    public void testInterruptedException() throws InterruptedException {
```

Comment:
Isn't this identical to [`TimeoutTest#failed`](https://github.com/jenkinsci/workflow-support-plugin/blob/e2ba3ca65dfe057aeb8986b7e1f114051e513264/src/test/java/org/jenkinsci/plugins/workflow/support/concurrent/TimeoutTest.java#L49) other than using `@Test(expected = ...)` instead of manually catching the exception? It doesn't seem useful to have both tests, so it seems better not to add this one.